app/api/routers/breeze_buddy/templates/handlers.py

```
@@ -297,3 +301,131 @@ async def get_template_by_id_handler(template_id: str, current_user: UserInfo):
+    )
+
+    try:
+        # First, get the existing template to validate access
+        existing_template = await get_template_by_id(template_id)
+
+        if not existing_template:
+            logger.warning(f"Template not found: {template_id}")
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"Template not found: {template_id}",
+            )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)